api/pkg/backend/service.go

```diff
@@ -205,17 +204,35 @@ func (s *APIService) CreateBackend(request *restful.Request, response *restful.R
 	actx := request.Attribute(c.KContext).(*c.Context)
 	backendDetail.TenantId = actx.TenantId
 	backendDetail.UserId = actx.UserId
+
+	storageTypes,_:= s.listStorageType(ctx,request,response)
+
+	_, foundType := typeExists(storageTypes.Types, backendDetail.Type)
+	if !foundType {
+		log.Errorf("Failed to retrieve backend type: %v\n", err)
+		response.WriteError(http.StatusBadRequest, err)
+		return
+	}
 	res, err := s.backendClient.CreateBackend(ctx, &backend.CreateBackendRequest{Backend: backendDetail})
 	if err != nil {
-		log.Errorf("failed to create backend: %v\n", err)
+		log.Errorf("Failed to create backend: %v\n", err)
 		response.WriteError(http.StatusInternalServerError, err)
 		return
 	}
 
-	log.Info("Create backend successfully.")
+	log.Info("Created backend successfully.")
 	response.WriteEntity(res.Backend)
 }
 
+func typeExists(slice []*backend.TypeDetail, inputType string) (int, bool) {
```

Comment:
Do we need to return two parameters, Are we using index?
Only boolean is not sufficient?